duckduckgo/macos-browser
# Hide all error messages when the user successfully connects

Task/Issue URL: https://app.asana.com/0/0/1205909767958379/f

## Description

We've been getting several reports where users are seeing error messages even after successfully connecting to the VPN.

We had logic in place to prevent this - being connected means no issues for general users.  At some point that broke due to the unintentional removal of an observer.  This PR adds it back.

## Steps to test this PR

You can run these steps in `develop` and you'll see that in `develop` the error message is always shown to the user.

1. Change the following line to assign a constant string to `lastControllerErrorMessage`:

https://github.com/duckduckgo/macos-browser/blob/3721244fd4aa1f42907ee90ed618323cfda7db30/LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/StatusView/NetworkProtectionStatusViewModel.swift#L208

2. Run the app.  Make sure that the error is shown.
3. Make sure that when you connect to NetP the error is hidden.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
